### PR TITLE
Removes ability for research cyborgs to self-upgrade

### PR DESCRIPTION
**Pull request by DigitalSquirrel95**
Port of https://github.com/TS-Rogue-Star/Rogue-Star/pull/335

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -737,13 +737,17 @@
 				update_icon()
 			else
 				to_chat(user, span_filter_notice("[span_red("Access denied.")]"))
+				if(user == src) //RS edit: No self-unlocking.
+					to_chat(user, "<span class='filter_notice'>You are not allowed to manipulate your own lock.</span>")
```
Comment (Kashargul) [inline code/modules/mob/living/silicon/robot/robot.dm:741]: use of old spans is forbiden

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 2]
```diff
@@ -737,13 +737,17 @@
...
 
 	else if(istype(W, /obj/item/borg/upgrade/))
 		var/obj/item/borg/upgrade/U = W
 		if(!opened)
 			to_chat(usr, span_filter_notice("You must access the borgs internals!"))
 		else if(!src.module && U.require_module)
 			to_chat(usr, span_filter_notice("The borg must choose a module before it can be upgraded!"))
+		else if(user == src) //RS edit: No self-upgrading.
+			to_chat(user, "<span class='warning'>You lack the reach to be able to upgrade yourself.</span>")
```
Comment (Kashargul) [inline code/modules/mob/living/silicon/robot/robot.dm:750]: use of old spans is forbidden

**Review by Kashargul — CHANGES_REQUESTED** [review]

**Comment by liache** [comment]
This seems like an unnecessary nerf. Do we really want this?

**Comment by Mattjpark89** [comment]
On other servers where people are trying to fulfill antag objectives and whatnot, this change makes sense as it increases the difficulty of hacking or modifying a neutral borg (as they can no longer unlock themselves simply by being told to).
On this server though? I don't see a reason for it. Unless it's being abused in some way I can't think of.

**Comment by SpaceMan-Jim** [comment]
I certainly don't believe this is a very valid issue to justify removing abilities. If there IS a case off a sci unit swapping their module around after upgrading. I'd personally say its a moderation issue than anything else, feels more spite filled considering said individual also wanted to remove the self repair.

**Pull request merged**